### PR TITLE
SBD Integration

### PR DESCRIPTION
**Pull request by neonphog**
DEPENDS ON:
- [x] https://github.com/holochain/holochain/issues/3479
- [x] https://github.com/holochain/holochain-infra/issues/96

TODO:
- [x] update "signal server" test config to match the production holochain sbd server to be created
- [x] update all examples and defaults of webrtc_config to match the production holochain stun servers to be created

### CODE REVIEW

File: crates/holochain_conductor_api/src/config/conductor.rs [hunk 1]
```diff
@@ -252,6 +252,12 @@ mod tests {
       transport_pool:
         - type: webrtc
           signal_url: wss://signal.holotest.net
+          webrtc_config: {
+            "iceServers": [
+              { "urls": "stun:stun.l.google.com:19302" },
+              { "urls": "stun:stun2.l.google.com:19302" }
+            ]
+          }
```
Comment (neonphog) [inline crates/holochain_conductor_api/src/config/conductor.rs:260]: TODO - update these to match the sbd production server to be created and holochain production stun servers to be created.

File: crates/kitsune_p2p/kitsune_p2p/src/spawn/actor.rs [hunk 2]
```diff
@@ -23,6 +23,15 @@ use std::collections::hash_map::Entry;
...
 
+/// Default webrtc config if set to `None`.
+/// TODO - set this to holochain stun servers once they exist!
+const DEFAULT_WEBRTC_CONFIG: &str = r#"{
+  "iceServers": [
+    { "urls": "stun:stun.l.google.com:19302" },
+    { "urls": "stun:stun2.l.google.com:19302" }
+  ]
+}"#;
+
```
Comment (neonphog) [inline crates/kitsune_p2p/kitsune_p2p/src/spawn/actor.rs:34]: TODO - update this to match the holochain production stun servers to be created.

File: crates/hc_sandbox/src/cmds.rs [hunk 3]
```diff
@@ -80,6 +80,9 @@ pub enum NetworkType {
     WebRTC {
         /// URL to a holochain tx5 WebRTC signal server.
         signal_url: String,
+
+        /// Optional override webrtc peer connection config to use.
+        webrtc_config: Option<String>,
```
Comment (ThetaSinner) [inline crates/hc_sandbox/src/cmds.rs:85]: This is accepting JSON on the command line? Would probably prefer to pass a path to a config file

**Review by ThetaSinner — APPROVED** [review]

File: crates/hc_sandbox/src/cmds.rs (hunk 3 above)
Comment (neonphog) [reply crates/hc_sandbox/src/cmds.rs]: Done: https://github.com/holochain/holochain/pull/3842/commits/9e85527e9a3b0130d4a454fdff52fb68f2b5b5a5

File: crates/kitsune_p2p/kitsune_p2p/src/spawn/actor.rs (hunk 2 above)
Comment (neonphog) [reply crates/kitsune_p2p/kitsune_p2p/src/spawn/actor.rs]: done

File: crates/holochain_conductor_api/CHANGELOG.md [hunk 4]
```diff
@@ -7,6 +7,8 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/).
 
 ## \[Unreleased\]
 
+- *BREAKING* Updates holochain to use the new SBD server architecture for webrtc signaling. If you were previously running your own tx5-signal-srv server for signaling, you will need to switch to [sbd-server](https://crates.io/crates/sbd-server). If you were using the holo-provided `wss://signal.holo.host`, you must switch to `wss://sbd-0.main.infra.holo.host`. See the module level documentation at [/src/config/conductor.rs](/src/config/conductor.rs) for a commented example holochain conductor configuration file. [\#3842](https://github.com/holochain/holochain/pull/3842)
```
Comment (pdaoust) [inline crates/holochain_conductor_api/CHANGELOG.md:10]:
```suggestion
- *BREAKING* Updates holochain to use the new SBD server architecture for WebRTC signaling. If you were previously running your own tx5-signal-srv server for signaling, you will need to switch to [sbd-server](https://crates.io/crates/sbd-server). If you were using the holo-provided `wss://signal.holo.host`, you must switch to `wss://sbd-0.main.infra.holo.host`. See the module level documentation at [/src/config/conductor.rs](/src/config/conductor.rs) for a commented example holochain conductor configuration file. [\#3842](https://github.com/holochain/holochain/pull/3842)
```

File: crates/holochain_conductor_api/src/config/conductor.rs [hunk 5]
```diff
@@ -1,5 +1,57 @@
...
+//!
+//! #### Example minimum conductor config:
+//!
+//! ```rust
+//! let yaml = r#"---
+//!
+//! ## Configure the keystore to be used.
+//! keystore:
+//!
+//!   ## use an in-process keystore with default database location.
```
Comment (pdaoust) [inline crates/holochain_conductor_api/src/config/conductor.rs:12]:
```suggestion
//!   ## Use an in-process keystore with default database location.
```

File: crates/holochain_conductor_api/src/config/conductor.rs [hunk 6]
```diff
@@ -1,5 +1,57 @@
...
+//!   type: lair_server_in_proc
+//!
+//! ## Configure an admin websocket interface at a specific port.
```
Comment (pdaoust) [inline crates/holochain_conductor_api/src/config/conductor.rs:15]:
```suggestion
//! ## Configure an admin WebSocket interface at a specific port.
```

File: crates/holochain_conductor_api/src/config/conductor.rs [hunk 7]
```diff
@@ -1,5 +1,57 @@
...
+//! admin_interfaces:
+//!   - driver:
+//!       type: websocket
+//!       port: 1234
+//!       allowed_origins: "*"
+//!
+//! ## Configure the network.
+//! network:
+//!
+//!   ## Use the holo-provided default production bootstrap server.
```
Comment (pdaoust) [inline crates/holochain_conductor_api/src/config/conductor.rs:25]:
```suggestion
//!   ## Use the Holo-provided default production bootstrap server.
```

File: crates/holochain_conductor_api/src/config/conductor.rs [hunk 8]
```diff
@@ -1,5 +1,57 @@
...
+//!   bootstrap_service: https://bootstrap.holo.host
+//!
+//!   ## [deprecated] Please just include this in your config file for now.
```
Comment (pdaoust) [inline crates/holochain_conductor_api/src/config/conductor.rs:28]: @jost-s notes that this isn't deprecated and is still required.

File: crates/holochain_conductor_api/src/config/conductor.rs [hunk 9]
```diff
@@ -1,5 +1,57 @@
...
+//!   network_type: quic_bootstrap
+//!
+//!   ## Setup a specific network configuration..
+//!   transport_pool:
+//!     ## use webrtc
```
Comment (pdaoust) [inline crates/holochain_conductor_api/src/config/conductor.rs:33]:
```suggestion
//!     ## Use WebRTC, which is the only option for now.
```

Is this the only option? If not, just accept the capitalisation fixes.

File: crates/holochain_conductor_api/src/config/conductor.rs [hunk 10]
```diff
@@ -1,5 +1,57 @@
...
+//! ## Configure the network.
+//! network:
+//!
+//!   ## Use the holo-provided default production bootstrap server.
+//!   bootstrap_service: https://bootstrap.holo.host
+//!
+//!   ## [deprecated] Please just include this in your config file for now.
+//!   network_type: quic_bootstrap
+//!
+//!   ## Setup a specific network configuration..
```
Comment (pdaoust) [inline crates/holochain_conductor_api/src/config/conductor.rs:31]:
```suggestion
//!   ## Setup a specific network configuration.
```

File: crates/holochain_conductor_api/src/config/conductor.rs [hunk 11]
```diff
@@ -1,5 +1,57 @@
...
+//!   transport_pool:
+//!     ## use webrtc
+//!     - type: webrtc
+//!
+//!       ## Use the holo-provided default production sbd (signal) server.
```
Comment (pdaoust) [inline crates/holochain_conductor_api/src/config/conductor.rs:36]:
```suggestion
//!       ## Use the Holo-provided default production sbd (signal) server.
```

File: crates/holochain_conductor_api/src/config/conductor.rs [hunk 12]
```diff
@@ -1,5 +1,57 @@
...
+//!       ## `signal_url` is REQUIRED.
+//!       signal_url: wss://sbd-0.main.infra.holo.host
+//!
+//!       ## Override the default webrtc stun configuration.
```
Comment (pdaoust) [inline crates/holochain_conductor_api/src/config/conductor.rs:40]:
```suggestion
//!       ## Override the default WebRTC STUN configuration.
```

**Review by pdaoust — CHANGES_REQUESTED** [review]
mostly just punctuation and capitalisation; only two comments re: the content itself.

File: crates/holochain_conductor_api/src/config/conductor.rs (hunk 8 above)
Comment (neonphog) [reply crates/holochain_conductor_api/src/config/conductor.rs:28]:
```suggestion
//!   ## This currently has no effect on functionality but is required. Please just include as-is for now.
```

File: crates/holochain_conductor_api/src/config/conductor.rs (hunk 9 above)
Comment (neonphog) [reply crates/holochain_conductor_api/src/config/conductor.rs]: It's effectively the only option, so it makes sense to say that here.

File: crates/holochain_conductor_api/src/config/conductor.rs (hunk 8 above)
Comment (neonphog) [reply crates/holochain_conductor_api/src/config/conductor.rs]: Let me know if that change works : )

**Review by pdaoust — APPROVED** [review]
LGTM!

**Pull request merged**